internal/commands/vaultsecrets/integrations/delete.go

```diff
@@ -0,0 +1,121 @@
+	Profile *profile.Profile
+	Output  *format.Outputter
+	IO      iostreams.IOStreams
+
+	IntegrationName string
+	Type            string
+	Client          secret_service.ClientService
+	PreviewClient   preview_secret_service.ClientService
+}
+
+func NewCmdDelete(ctx *cmd.Context, runF func(*DeleteOpts) error) *cmd.Command {
+	opts := &DeleteOpts{
+		Ctx:           ctx.ShutdownCtx,
+		Profile:       ctx.Profile,
+		Output:        ctx.Output,
+		IO:            ctx.IO,
+		Client:        secret_service.New(ctx.HCP, nil),
+		PreviewClient: preview_secret_service.New(ctx.HCP, nil),
+	}
+
+	cmd := &cmd.Command{
+		Name:      "delete",
+		ShortHelp: "Delete a Vault Secrets integration.",
+		LongHelp: heredoc.New(ctx.IO).Must(`
+      The {{ template "mdCodeOrBold" "hcp vault-secrets integrations delete" }} command deletes a Vault Secrets integration.
+      `),
+		Examples: []cmd.Example{
+			{
+				Preamble: `Delete an integration:`,
+				Command: heredoc.New(ctx.IO, heredoc.WithPreserveNewlines()).Must(`
+            $ hcp vault-secrets integrations delete sample-integration --type twilio
+            `),
+			},
+		},
+		Args: cmd.PositionalArguments{
+			Args: []cmd.PositionalArgument{
+				{
+					Name:          "NAME",
+					Documentation: "The name of the integration to delete.",
+				},
+			},
+		},
+		Flags: cmd.Flags{
+			Local: []*cmd.Flag{
+				{
+					Name:         "type",
+					DisplayValue: "TYPE",
+					Description:  "The type of the integration to delete.",
+					Value:        flagvalue.Simple("", &opts.Type),
+					Required:     true,
+				},
+			},
+		},
+		RunF: func(c *cmd.Command, args []string) error {
+			opts.IntegrationName = args[0]
+
+			if runF != nil {
+				return runF(opts)
+			}
+			return deleteRun(opts)
+		},
+	}
+
+	return cmd
+}
+
+func deleteRun(opts *DeleteOpts) error {
+	switch opts.Type {
+	case Twilio:
+		_, err := opts.PreviewClient.DeleteTwilioIntegration(&preview_secret_service.DeleteTwilioIntegrationParams{
+			Context:         opts.Ctx,
+			ProjectID:       opts.Profile.ProjectID,
+			OrganizationID:  opts.Profile.OrganizationID,
+			IntegrationName: opts.IntegrationName,
+		}, nil)
+		if err != nil {
+			return fmt.Errorf("failed to delete integration: %w", err)
+		}
+
+		fmt.Fprintf(opts.IO.Err(), "%s Successfully deleted integration with name %q\n", opts.IO.ColorScheme().SuccessIcon(), opts.IntegrationName)
+		return nil
+
+	case MongoDB:
+		_, err := opts.PreviewClient.DeleteMongoDBAtlasIntegration(&preview_secret_service.DeleteMongoDBAtlasIntegrationParams{
+			Context:         opts.Ctx,
+			ProjectID:       opts.Profile.ProjectID,
+			OrganizationID:  opts.Profile.OrganizationID,
+			IntegrationName: opts.IntegrationName,
+		}, nil)
+		if err != nil {
+			return fmt.Errorf("failed to delete integration: %w", err)
+		}
+
+		fmt.Fprintf(opts.IO.Err(), "%s Successfully deleted integration with name %q\n", opts.IO.ColorScheme().SuccessIcon(), opts.IntegrationName)
+		return nil
+
+	default:
```

Comment:
It may also be good to support "aws" and "gcp" types here, they follow identical pattern to rotation integrations.

Reply:
@averche Are those considered "generic integrations" as well? I was looking at the providers supported for adding an integration for rotating secrets (only saw mongodb and twilio), but I can probably use a bit more clarity on the classification of "aws" and "gcp" types here as they relate to type of integrations. Seems like there are also aws and gcp sync integrations?

Reply:
The AWS and GCP integrations that were added to support dynamic secrets are the "new-style" integrations very similar to Twilio and MongoDB Atlas. Unlike the older sync integrations, these 4 new integrations are:
- stored in the `third_party_integrations` table
- support multiple capabilities (AWS and GCP support both "ROTATION" and "DYNAMIC" capabilities)
- follow the same CRUDL pattern under `/org/{org}/project/{proj}/integrations` and `/org/{org}/project/{proj}/integrations/{provider}/config`

We are in the process of adding functionality to reuse AWS integrations for supporting rotating secrets. GCP rotating secrets will be added next sprint.

Reply:
This is super thorough - thanks so much 🙌. I'll add those 2 providers 👌.